models/tshirt/include/Tshirt.h

```diff
@@ -0,0 +1,289 @@
+
+namespace tshirt {
+
+    //! Tshirt parameters struct
+    /*!
+        This structure provides storage for the parameters of the Tshirt hydrological model
+    */
+
+    struct tshirt_params
+    {
+        double maxsmc;              //!< saturated soil moisture content (sometimes theta_e)
+        double wltsmc;              //!< wilting point soil moisture content
+        double satdk;               //!< vertical saturated hydraulic conductivity [m s^-1] (sometimes Kperc or Ks)
+        double satpsi;              //!< saturated capillary head [m]
+        // TODO: explain more what this is
+        double slope;               //!< SLOPE parameter
+        double b;                   //!< 'b' exponent on Clapp-Hornberger soil water relations (sometime bexp)
+        double multiplier;          //!< the multiplier applied to 'satdk' to route water rapidly downslope in subsurface (sometimes 'mult' or 'LKSATFAC')
+        double alpha_fc;            //!< alpha constant for given soil type for relative suction head value, with respect to Hatm
+        double Klf;                 //!< lateral flow independent calibration parameter
+        double Kn;                  //!< Nash cascade linear reservoir coefficient lateral flow parameter
+        int nash_n;                 //!< number of nash cascades
+        double Cgw;                 //!< Ground water flow param
+        double expon;               //!< Ground water flow exponent param (analogous to NWM 2.0 expon param)
+        double Ssmax;               //!< Subsurface soil water flow max storage param (calculated from maxsmc and depth)
+        double Sgwmax;              //!< Ground water flow max storage param (analogous to NWM 2.0 zmax param)
+        const double depth = 2.0;         //!< Total soil column depth ('D') [m]
+
+        //! Constructor for tshirt parameters
+        /*!
+            Constructor for tshirt param objects.
+        */
+        tshirt_params(double maxsmc, double wltsmc, double satdk, double satpsi, double slope, double b,
+                double multiplier, double alpha_fc, double Klf, double Kn, int nash_n, double Cgw, double expon) :
+                maxsmc(maxsmc),
+                wltsmc(wltsmc),
+                satdk(satdk),
+                satpsi(satpsi),
+                slope(slope),
+                b(b),
+                multiplier(multiplier),
+                alpha_fc(alpha_fc),
+                Klf(Klf),
+                Kn(Kn),
+                nash_n(nash_n),
+                Cgw(Cgw),
+                expon(expon)
+        {
+            this->Ssmax = this->depth * this->maxsmc;
+        }
+
+    };
+
+    /*!
+     * Tshirt state structure
+     *
+     * This structure provides storage for the state used by Tshirt hydrological model at a particular time step.
+     */
+    struct tshirt_state
+    {
+        // TODO: confirm this is correct
+        double Ss;           //!< current water storage in soil column nonlinear reservoir
```

Comment:
ultimately would like to see these follow a more verbose convention, i.e. soil_storage_meters, ground_water_storage_meters, nash_cascade_storeage_meters (or whaterver appropriate units these variables are modeling)

Reply:
I have renamed most of these, but there are a handful (e.g., `Klf`) I haven't.  I think it makes sense for certain constant parameters to be left like this and/or reflect external conventions, but I won't mark this conversation resolved just yet.